doc/articles/Extensions.md

```diff
@@ -0,0 +1,71 @@
+# Extensions {#Extensions}
+
+`webgpu.h` is designed to be an extensible and forward-compatible API.
+The following types of extensions are supported:
+
+```
+wgpuPrefixNewFunction
+
+WGPUPrefixNewObject
+wgpuPrefixNewObjectNewMethod
+wgpuOldObjectPrefixNewMethod
+
+WGPUPrefixNewEnum
+WGPUPrefixNewEnum_NewValue
+WGPUOldEnum_PrefixNewValue
+
+WGPUPrefixNewStruct
+
+WGPUPrefixNewBitflagType
+WGPUPrefixNewBitflagType_NewValue
+WGPUOldBitflagType_PrefixNewValue
+
+WGPUPrefixNewCallback
+
+WGPU_PREFIX_NEW_CONSTANT
+```
+
+("Prefix" is the name of the implementation that owns the extension, if any; see below.)
+
+When an application is running against an unknown `webgpu.h` implementation, extension support may be detected at runtime as follows:
+
+- New functions/methods may be detected via @ref wgpuGetProcAddress().
+    - New objects may be detected by the presence of the methods that create them.
+    - New (root) structs and callback types are only relevant if the methods that accept them exist.
+- New enums, bit flags, [chained structs](@ref StructChaining) are available iff enabled. Like other features, these must be both detected and enabled:
+    - Device features are detected via @ref wgpuAdapterGetFeatures() and enabled via @ref WGPUDeviceDescriptor::requiredFeatures.
+    - Instance features are detected via @ref wgpuGetInstanceCapabilities() and enabled via @ref WGPUInstanceDescriptor::capabilities.
+
+The following design principles should be followed to ensure future extensibility:
+
+- Enums always have a `Force32 = 0x7FFFFFFF` value to force them to be 32-bit (and have a stable ABI representation).
+- Bitflag types are always 64-bit.
+- Structures should be extensible (have a `nextInChain`), or at least be associated with some struct (e.g. child, sibling, or parent) that is extensible.
+
+Note also:
+
+- Whenever necessary a version `2` or implementation-specific version of an existing method or type can be added.
+
+## Registry of prefixes and enum blocks
+
+Implementation extensions **should**, and multi-implementation extensions **must**, use the naming conventions listed above with the prefixes listed here.
+
+Implementation-specific and multi-implementation extensions **must** not use block `0x0000_????` when adding values to existing enums.
+
+|                    | Prefix       | Enum Block    | Description
+|--------------------|--------------|---------------|------------
+| Core               | (none)       | `0x0000_????` | Extensions that are required to implement (e.g. added after 1.0).
+| Multi-Vendor       | (none)       | `0x0001_????` | Extensions that are optional to implement (e.g. platform-specific extensions). *TBD if enum values in this block may become required, or if they would be aliased into block `0x0000_????` first.*
```

Comment:
There is a question buried here which I noticed in #417 and then completely forgot about.

> Should the new enums really use block 0x0001? Seems like if we ever put this stuff in "core" they could use the same enums, would we then duplicate them into block 0x0000 or just use the block 0x0001 values? Or should we just put them in block 0x0000 to start?

Ugh. Can't keep track of all this minutiae.

Reply:
@cwfitzgerald any thoughts? (no rush, I'm ~out until Jan)

Reply:
Maybe stupid question... but would it be possible to just renumber an enum if it gets promoted? Would that be a breaking change? (I guess it COULD be if someone is using the actual value of the enum or if we have different header versions in a wire situation?) I'm just wondering how much breakage it could cause and whether we could just promote them and say here that we may do that so don't rely on the actual value of the enums.